tls/s2n_connection.h

```diff
@@ -305,6 +305,14 @@ struct s2n_connection {
 
     /* Key update data */
     unsigned key_update_pending:1;
+
+    /* Bitmap to represent preferred list of keyshare for client to generate and send keyshares in the ClientHello message.
+     * The least significant bit (lsb), if set, indicates that the client must send an empty keyshare list.
+     * Each bit value in the bitmap indiciates the corresponding curve in the ecc_preferences list for which a key share needs to be generated.
+     * The order of the curves represented in the bitmap is obtained from the security_policy->ecc_preferences.
+     * Setting and manipulating this value requires security_policy to be configured prior.
+     * */
+    uint8_t preferred_key_shares;
```

Comment:
This is also an option, although probably overkill since you're dealing with such a small number of flags: https://github.com/awslabs/s2n/blob/master/utils/s2n_bitmap.h

Reply:
Yeah, this might be a bit overkill for my use-case.